surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,235 @@
+- `cargo clippy`
+- `cargo test`
+- `cargo run` / `cargo build`
+
+Frequency:
+
+- After every code change
+- Only after I have resolved other issues
+- Only after I am done (e.g. before creating a commit or pushing to a remote branch)
+- Never
+
+### How long do you need to wait for the compiler to rebuild your code after making a change?
+
+Please select the longest duration range amongst the projects that you regularly work on.
+
+This question focuses on local development workflows, not CI or remote builds. If you do not compile your Rust code locally, please skip this question.
+
+Type: select one (optional)
+
+- Less than a second
+- Between 1 and 5 seconds
+- Between 5 and 10 seconds
+- Between 10 and 30 seconds
+- Between 30 seconds and 1 minute
+- Between 1 and 5 minutes
+- More than 5 minutes
+
+### Which of the following problems do you most struggle with?
+
+Type: matrix (optional)
+
+Commands:
+
+- Waiting for a rebuild after making a small change
+- Waiting for CI workflows that build Rust code
+- `cargo check` and `cargo build` not sharing compilation cache
+- `cargo check` and `cargo clippy` not sharing compilation cache
+- Waiting for an IDE to show me inline error/warning annotations
+- `cargo` rebuilding everything from scratch and I do not understand why
+
+Priority:
+
+- Not a problem for me
+- Could be improved
+- Big blocker for me
+
+## Workarounds
+
+### Have you used any of the following mechanisms to improve compilation performance?
+
+Please select only mechanisms/workarounds that you have used at least once **primarily** in order to improve compilation performance, not for other reasons.
+
+Type: select all that apply (optional)
+
+- Disable (or reduce) debuginfo (e.g. set `debug = 0` in `Cargo.toml`)
+- Parallel compiler frontend (pass `-Zthreads=<N>` to the compiler)
+- Cranelift codegen backend (e.g. set `codegen-backend = "cranelift"` in `Cargo.toml`)
+- Alternative linker (e.g. `lld`/`mold`/`wild`)
+- Caching compiler wrapper (e.g. `sccache`)
+- Share `target` directory amongst multiple projects (with `CARGO_TARGET_DIR`)
+- Split crates into smaller crates
+- Reduce the amount of dependencies
+- Disable default Cargo features of dependencies
+- Create a Cargo feature that makes certain dependencies (or their features) optional
+- Reduce usage of procedural macros
+- Reduce usage of generic code (e.g. by converting it to use `dyn Trait` instead)
+- Merge integration tests into a single binary
+- Something else (open response)
+
+### If you use an alternative linker, which one do you use?
+
+Type: select all that apply (optional)
+
+- gold
+- lld
+- mold
+- wild
+- Other (open response)
+
+### Do you use a global `config.toml` override?
+
+You can create a `config.toml` file in your `CARGO_HOME` directory (e.g. `~/.cargo/config.toml`) which can be used to apply certain compilation settings (e.g. using a faster linker) across all Cargo projects on your computer.
+
+Type: select one (optional)
+
+- Yes
+- No
```

Comment:
Should we ask why not?  I know for me, I regularly run into problems with MSRV and my global config.